pinot-core/src/main/java/org/apache/pinot/core/upsert/RecordLocation.java

```diff
@@ -28,13 +28,13 @@
   private final String _segmentName;
   private final int _docId;
   private final long _timestamp;
-  private final boolean _isConsuming;
+  private final ThreadSafeMutableRoaringBitmap _validDocIds;
 
-  public RecordLocation(String segmentName, int docId, long timestamp, boolean isConsuming) {
+  public RecordLocation(String segmentName, int docId, long timestamp, ThreadSafeMutableRoaringBitmap validDocIds) {
```

Comment:
not sure if `ThreadSafeMutableRoaringBitmap ` is the best identifier of the containing segment. Perhaps the segmentImpl itself, in case `ThreadSafeMutableRoaringBitmap ` itself may be replaced?

Reply:
I don't want to put too many unrelated info here. If we decide to use another data structure, we can change the constructor. This class is just a wrapper, not an interface, so changing constructor should be fine.

Reply:
sgtm